FreeRTOS/FreeRTOS-Plus-TCP
# Fix assertion failure when disconnecting the Ethernet cable (NXP1060 -  NetworkInterface.c)

The changes that I'm proposing are to make the driver to work in case of cable disconnection. 


Description
-----------
In prvEMACHandlerTask, when xSendEventStructToIPTask, we have to pass network interface or it will fail the assertion in xSendEventStructToIPTask "configASSERT( pxInterface != NULL )".

In pxNXP1064_NetworkInterfaceInitialise, if xWaitPHY fails (for instance, the cable is not connected), the switch case will break without doing nothing and the prvEMACHandlerTask will be blocked waiting for the notification. We have to unblock it to try again in case of failure in xEMAC_WaitPHY.

Test Steps
-----------
With the original code, connect the Ethernet  cable, let the DHCP make what it has to make, disconnect the cable and you will get the following message:
Link down!
ASSERT ERROR " pxInterface != ((void *)0) ": file "../FreeRTOS_TCP/Source/FreeRTOS_IP_Utils.c" Line "823" function name "prvProcessNetworkDownEvent" 


Apply the fix and you will get:
Link down!
Network Down
Configuration successful.
Waiting for link to go up and auto-negotiation to complete.
Failed to get the link up.
Link down!
Network Down
Configuration successful.
Waiting for link to go up and auto-negotiation to complete.
Link up.Waiting for auto-negotiation to complete.
Auto-negotiation complete.



Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
